macos/FluentUI/Button/Button.swift

```diff
@@ -433,11 +489,62 @@ open class Button: NSButton {
 			guard oldValue != style else {
 				return
 			}
+			setupBorderShadowsIfNeeded()
 			setColorValues(forStyle: style, accentColor: accentColor)
 			needsDisplay = true
 		}
 	}
 
+	var usesBorderShadows: Bool = false {
+		didSet {
+			guard oldValue != usesBorderShadows else {
+				return
+			}
+
+			if usesBorderShadows {
+				firstOuterDropShadowLayer = CALayer()
+				secondOuterDropShadowLayer = CALayer()
+				innerShadowLayer = CALayer()
+
+				// Ensure that the Shadow sublayers can in fact extend beyond the bounds of the Button View's
+				// layer so that the shadows don't get clipped
+				self.layer?.masksToBounds = false
+
+				if let firstOuterDropShadowLayer = firstOuterDropShadowLayer {
```

Comment:
can create local (non-optional) variables with a different name and then assign to firstOuterDropShadowLayer at the end of this func to avoid this if lets when we know the object is nonnull